force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,44 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+    if (Trigger.isBefore && Trigger.isUpdate){
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity opp : Trigger.New){
+            if (opp.AccountId != null) {
+                accountIds.add(opp.AccountId);
+            }
+            if(opp.Amount <= 5000 && opp.Amount != null){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+        Map<Id, Contact> accountToCeoContact = new Map<Id, Contact>();
+        if (!accountIds.isEmpty()) {
+            for (Contact con : [SELECT Id, AccountId FROM Contact WHERE Title = 'CEO' AND AccountId IN :accountIds]) {
+                accountToCeoContact.put(con.AccountId, con);
+            }
+        }
+        
+        for (Opportunity opp : Trigger.new) {
+            if (opp.AccountId != null && accountToCeoContact.containsKey(opp.AccountId)) {
+                opp.Primary_Contact__c = accountToCeoContact.get(opp.AccountId).Id;
```

Comment:
Try to check  if the primary contact is null before updating it, you might not need to if it's already populated